src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicSignatureHelp.cs

```diff
@@ -0,0 +1,308 @@
+    ''' </summary>
+    ''' <param name=""i"">an integer, preferably 42.</param>
+    ''' <param name=""i2"">an integer, anything you like.</param>
+    ''' <returns>returns an object of type C</returns>
+    Function Method(i As Integer, i2 As Integer) As C
+        Return Nothing
+    End Function
+
+
+    ''' <summary>
+    ''' Hello Generic World!
+    ''' </summary>
+    ''' <typeparam name=""T1"">Type Param 1</typeparam>
+    ''' <param name=""i"">Param 1 of type T1</param>
+    ''' <returns>Null</returns>
+    Function GenericMethod(Of T1)(i As T1) As C
+        Return Nothing
+    End Function
+
+
+    Function GenericMethod(Of T1, T2)(i As T1, i2 As T2) As C
+        Return Nothing
+    End Function
+
+
+    ''' <summary>
+    ''' Complex Method Params
+    ''' </summary>
+    ''' <param name=""strings"">Jagged MultiDimensional Array</param>
+    ''' <param name=""outArr"">Out Array</param>
+    ''' <param name=""d"">Dynamic and Params param</param>
+    ''' <returns>Null</returns>
+    Sub OutAndParam(ByRef strings As String()(,), ByRef outArr As String(), ParamArray d As Object)
+    End Sub
+End Class");
+
+            SendKeys("Dim gm = GenericMethod");
+            SendKeys(VirtualKey.Escape);
+            SendKeys("(");
+            VerifyCurrentSignature("C.GenericMethod(Of T1)(i As T1) As C\r\nHello Generic World!");
+            VerifyCurrentParameter("i", "Param 1 of type T1");
+            VerifyParameters(
+                ("i", "Param 1 of type T1"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.SignatureHelp)]
+        public void GenericMethodSignatureHelp2()
+        {
+            SetUpEditor(@"
+Imports System
+Class C(Of T, R)
+    Sub M()
+        $$
+    End Sub
+    
+    ''' <summary>
+    ''' Generic Method with 1 Type Param
+    ''' </summary>
+    ''' <typeparam name=""T1"">Type Parameter</typeparam>
+    ''' <param name=""i"">param i of type T1</param>
+    Sub GenericMethod(Of T1)(i As T1)
+    End Sub
+
+
+    ''' <summary>
+    ''' Generic Method with 2 Type Params
+    ''' </summary>
+    ''' <typeparam name=""T1"">Type Parameter 1</typeparam>
+    ''' <typeparam name=""T2"">Type Parameter 2</typeparam>
+    ''' <param name=""i"">param i of type T1</param>
+    ''' <param name=""i2"">param i2 of type T2</param>
+    ''' <returns>Null</returns>
+    Function GenericMethod(Of T1, T2)(i As T1, i2 As T2) As C(Of T, R)
+        Return Nothing
+    End Function
+End Class");
+
+            SendKeys("GenericMethod");
+            SendKeys(VirtualKey.Escape);
+            SendKeys("(Of ");
+            VerifyCurrentSignature("C(Of T, R).GenericMethod(Of T1)(i As T1)\r\nGeneric Method with 1 Type Param");
+            VerifyCurrentParameter("T1", "Type Parameter");
+            VerifyParameters(
+                ("T1", "Type Parameter"));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.SignatureHelp)]
+        public void GenericMethodSignatureHelp_CtrlKP()
```

Comment:
The name of this test implies that it's purpose is to test via a different invocation path, but it doesn't do anything different than the others.